src/sentry/rules/conditions/event_frequency.py

```diff
@@ -278,6 +284,39 @@ def query_hook(
         )
         return sums[event.group_id]
 
+    def get_sums(
+        self, model: int, groups: list[int], start: datetime, end: datetime, environment_id: int
+    ) -> dict[int, int]:
+        sums: Mapping[int, int] = self.tsdb.get_sums(
+            model=get_issue_tsdb_group_model(model),
+            keys=[group.id for group in groups],
+            start=start,
+            end=end,
+            environment_id=environment_id,
+            use_cache=True,
+            jitter_value=groups[0].id,
+            tenant_ids={"organization_id": groups[0].project.organization_id},
+            referrer_suffix="alert_event_frequency",
+        )
+        return sums
+
+    def batch_query_hook(
+        self, group_ids: list[int], start: datetime, end: datetime, environment_id: str
+    ) -> dict[int, int]:
+        batch_sums: dict[int, int] = defaultdict(int)
+        groups = Group.objects.filter(id__in=group_ids)
+        error_issues = [group for group in groups if group.issue_category == GroupCategory.ERROR]
```

Comment:
I broke these out because it seems to matter which model we pass to the query, but it's only a distinction between error issues and everything else.

https://github.com/getsentry/sentry/blob/master/src/sentry/issues/constants.py#L14-L15

Reply:
Two kinda separate questions:
1. Do we need to do this for every batch_query_hook or just the `EventFrequencyCondition`?

1. Is there anyway that we could change / improve the query to still allow us to run in a single scan? Since this will create two scans, one for each chunk, it will cut our potential wins for the project in half. (This is still a big win that N scans < 2 scans)

Reply:
Discussed off GH but snuba has this stuff in different datasets, so it'll have to be this way for all the batched conditions. It's ok though since the vast majority of events are in issues so we're still saving a huge number of queries by batching.